### PR TITLE
Correlations: Add query editor and target field to settings page

### LINKED ISSUE #55369: Glue: Query Editor for target query
Create new UI elements displayed when adding or editing a correlation entry:
* Target Field
* Target Query Editor

Screen Shot 2022-09-19 at 12 15 54

When a correlation is added/edited the _target field_ and _target query_ should be saved using new API methods defined in #55368.

The editor should be reloaded when target data source is changed.

Target field is a static text input (once we get results from a sample source query it will be changed to a dropdown).

### PR DESCRIPTION
**Pull request by Elfo404**
Adds a target field and a query editor to correlations page page.

Fixes #55369, waiting for ~https://github.com/grafana/grafana/pull/55560~ https://github.com/grafana/grafana/issues/55376 (the PR works without it, it just doesn't allow editing correlations config)

### CODE REVIEW

File: public/app/features/correlations/Forms/QueryEditorField.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    value: datasource,
+    loading: dsLoading,
+    error: dsError,
+  } = useAsync(async () => {
+    if (!dsUid) {
+      return;
+    }
+    return getDataSourceSrv().get(dsUid);
+  }, [dsUid]);
+  const QuerEditor = datasource?.components?.QueryEditor;
```
Comment (ifrost) [inline public/app/features/correlations/Forms/QueryEditorField.tsx:26]: Small typo `QuerEditor` -> `QueryEditor`

File: public/app/features/correlations/Forms/QueryEditorField.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  return (
+    <Field label="Query" invalid={invalid} error={error}>
+      <Controller
+        name={name}
+        // FIXME: this won't be needed when the API will return the default config
```
Comment (ifrost) [inline public/app/features/correlations/Forms/QueryEditorField.tsx:32]: I would prefer to cross-reference it to make sure it's not forgotten. I've created a clean-up issue: https://github.com/grafana/grafana/issues/55737
```suggestion
        // TODO: #55737 this won't be needed when the API will return the default config
```

File: packages/grafana-ui/src/components/TextArea/TextArea.tsx [hunk 3]
```diff
@@ -24,6 +24,7 @@ const getTextAreaStyle = stylesFactory((theme: GrafanaTheme2, invalid = false) =
       sharedInputStyle(theme),
       getFocusStyle(theme),
       css`
+        display: block;
```
Comment (ifrost) [inline packages/grafana-ui/src/components/TextArea/TextArea.tsx:27]: It changes `inline-block` to `block` - are we sure it won't affect any existing components? If not, would it be better to change CSS for our use case only and create a separate issue just to change `display: block`?

File: public/app/features/correlations/Forms/types.ts [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
   targetUID: string;
   label: string;
   description: string;
+  config: CorrelationConfig;
 }
 
-type FormDTOWithoutTarget = Omit<FormDTO, 'targetUID'>;
-export type EditFormDTO = Partial<FormDTOWithoutTarget> & Pick<FormDTO, 'sourceUID'> & { uid: Correlation['uid'] };
+type FormDTOWithoutTarget = Omit<FormDTO, 'targetUID' | 'sourceUID'>;
+export type EditFormDTO = FormDTOWithoutTarget;
```
Comment (ifrost) [inline public/app/features/correlations/Forms/types.ts:12]: It's much simpler so probably can be inlined now.

```suggestion
export type EditFormDTO = Omit<FormDTO, 'targetUID' | 'sourceUID'>;
```

**Review by ifrost — COMMENTED** [review]
This is so sweeeet 🎉

**Review by L-M-K-B — APPROVED** [review]
Looks ok to me! 👍

**Review by ifrost — APPROVED** [review]
so-beautiful-shia-la-beouf

**Pull request merged**